Assets/Scripts/CharacterControllers/Melody/States/DodgeState.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace Melody.States
+{
+    using UnityEngine;
+
+    public class DodgeState : MelodyState
+    {
+
+        protected Vector3 Dodge = Vector3.left;
+
+        public DodgeState(MelodyController controller) : base(controller)
+        {
+        }
+
+        protected override void Enter()
+        {
+            melodyController.animator.SetTrigger("Dodge");
+            melodyController.rigidBody.useGravity = false;
+            Dodge = new Vector3(1 * Mathf.Sin(Mathf.Deg2Rad * melodyController.transform.eulerAngles.y), 0, Mathf.Cos(Mathf.Deg2Rad * melodyController.transform.eulerAngles.y));
+            Dodge *= 4;
```

Comment:
Should we perhaps store this dodge multiplier somewhere in the MelodyController and grab it from there? I'd prefer to keep Melody's movement values all in one place to make it easy to tweak things.